utils/hare-fetch-fids

```diff
@@ -0,0 +1,191 @@
+
+import sys
+import re
+from typing import List, NamedTuple
+import json
+import argparse
+import subprocess
+import simplejson
+from hax.types import ObjT
+
+
+class Node:
+    def __init__(self, name: str):
+        self.name = name
+        self.svcs = []
+
+    def add_service(self, svc: str, fid: str):
+        self.svcs.append(Process(svc, fid))
+
+    def get_service(self, service: str):
+        for svc in self.svcs:
+            if svc.name == service:
+                return svc.fid
+        print('Service not found', file=sys.stderr)
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}(name={self.name}, svcs={self.svcs})'
+
+    def for_json(self):
+        return self.__dict__
+
+
+Process = NamedTuple('Process', [('name', str), ('fid', str)])
+
+
+def to_fid(container: int, key: int) -> str:
+    return f'{container:#x}:{key:#x}'
+
+
+def node_names(data: List) -> List[Node]:
+    # Get entries from file
+    # ​"key": "m0conf/nodes/<node_fid>",
+    # "value": "{​​​​​\"name\": \"<node name>\", \"state\": \"<HA state>\"}​​​​​"
+    regex = re.compile(
+        f'm0conf/nodes/[a-zA-Z0-9:]+$')
+    nodes = []
+    for key in data:
+        match_result = re.match(regex, key['key'])
+        if not match_result:
+            continue
+        if len(key['key'].split('/')) == 3:
+            nodes.append(json.loads(key['value'])['name'])
+    return nodes
+
+
+def get_keys_for_processes(data: List) -> List[str]:
+    # Get 'm0conf/nodes/<node_id>/processes/<process_fidk>/services/<svc_type>' entries
+    # from the file.  See 'Consul KV Schema' in [4/KV](rfc/4/README.md).
+    regex = re.compile(
+        f'^m0conf\\/.*\\/processes\\/.*\\/services\\/(.+)$')
+    keys = []
+    for key in data:
+        match_result = re.match(regex, key['key'])
+        if not match_result:
+            continue
+        keys.append(match_result.group(0))
+    return keys
+
+
+def add_service_to_node(nodes: List[Node], node_name: str, svc_name: str, fid: str):
+    for node in nodes:
+        if node.name == node_name:
+            node.add_service(svc_name, fid)
+
+
+def get_service_for_node(nodes: List[Node], node_name: str = None, service: str = None) -> str:
+    # if node_name is not specified, then local node is used.
+    if node_name == None:
+        process = subprocess.run("node-name",
+                                check=True,
+                                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        str = process.stdout.strip()
+        node_name = str.decode('utf-8')
+
+    # if service is not specified, return all the services for a node.
+    if service == None:
+        for node in nodes:
+            if node.name == node_name:
+                return simplejson.dumps(node, indent=2, for_json=True)
+
+    for node in nodes:
+        if node.name == node_name:
+            return node.get_service(service)
+
+    print('Node-name not found in cluster.', file=sys.stderr)
```

Comment:
print(f'Node {node_name} not found', file=sys.stderr) please.

Reply:
done